src/syrupy/extensions/json.py

```diff
@@ -0,0 +1,81 @@
+import datetime
+import json
+from gettext import gettext
+from pathlib import Path
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Optional,
+)
+
+from syrupy.data import SnapshotFossil
+from syrupy.extensions.amber.serializer import Repr
+from syrupy.extensions.single_file import SingleFileSnapshotExtension
+
+if TYPE_CHECKING:
+    from syrupy.types import (
+        PropertyFilter,
+        PropertyMatcher,
+        PropertyPath,
+        SerializableData,
+    )
+
+
+class JSONSnapshotExtension(SingleFileSnapshotExtension):
+    @property
+    def _file_extension(self) -> str:
+        return "json"
+
+    @classmethod
+    def _filter(
+        cls,
+        data: SerializableData,
+        filtered: SerializableData,
+        path: PropertyPath,
+        exclude: Optional["PropertyFilter"] = None,
+        matcher: Optional[PropertyMatcher] = None,
+    ) -> SerializableData:
+        if not isinstance(data, dict):
```

Comment:
I've refactored this function a little bit. I've tested it with exclude(props and paths) and matcher path_type - and it seems to work. What do you think?
```
        if not isinstance(data, dict):
            return data
        for k, v in data.items():
            if exclude and exclude(prop=k, path=path):
                continue
            path_tpl = (*path, (k, type(v)))
            if isinstance(v, dict):
                filtered[k] = {}
                filtered[k] = cls._filter(v, filtered[k], path_tpl, exclude, matcher)
                continue
            elif isinstance(v, list):
                filtered[k] = []
                for item in v:
                    result = cls._filter(item, {}, path_tpl, exclude, matcher)
                    filtered[k].append(result)
                continue
            if matcher:
                v = matcher(data=v, path=path_tpl)
            if isinstance(v, Repr):
                filtered[k] = repr(v)
            elif isinstance(v, datetime.datetime):
                filtered[k] = v.strftime("%Y-%m-%dT%H:%M:%S.%f%z")
            else:
                filtered[k] = v
        return filtered
```